modules/cold-extraction/ColdDataRetriever.py

```diff
@@ -305,16 +278,133 @@ def run_threaded(job_func):
     job_thread = threading.Thread(target=job_func)
     job_thread.start()
     
-
-# The thread scheduling
-schedule.every(1).minutes.do(run_threaded, run_retrieval)    
-schedule.every(10).minutes.do(run_threaded, update_pickle)
-
-# Keep running in a loop.
-while True:
+def run_cold_extraction():
+    read_csv()
+    # The thread scheduling
+    schedule.every(1).minutes.do(run_threaded, run_retrieval)    
+    schedule.every(10).minutes.do(run_threaded, update_pickle)
+
+    # # Keep running in a loop.
+    while True:
+        try:
+            schedule.run_pending()
+            time.sleep(1)
+        except KeyboardInterrupt:
+            check_kill_process()
+            logging.shutdown()
+            sys.exit(0)
+
+if __name__ == "__main__":
+    global storescp_processes, niffler_processes, nifflerscp_str, qbniffler_str
+    global storage_folder, file_path, csv_file, extraction_type, accession_index, patient_index, date_index, date_type, date_format, email, send_email
+    global DCM4CHE_BIN, SRC_AET, QUERY_AET, DEST_AET, NIGHTLY_ONLY, START_HOUR, END_HOUR, IS_EXTRACTION_NOT_RUNNING, NIFFLER_ID, MAX_PROCESSES, SEPARATOR
+    global accessions, patients, dates, niffler_log, resume, upload_folder
+
+    upload_folder = './csv/'
```

Comment:
Like I explained above, it is not necessary to separate the upload_folder in the cold-extraction module.

Just pass on the value for "csv_file" from the front-end module.